tablo/interfaces/arcgis/views.py

```diff
@@ -202,60 +212,83 @@ def handle_request(self, request, **kwargs):
-            search_params['only_return_count'] = True
-        elif kwargs.get('returnGeometry', 'true') == 'true':
-            search_params['return_geometry'] = True
+            search_params['return_geometry'] = False
+        elif return_count_only:
+            search_params['count_only'] = True
+        elif kwargs.get('returnGeometry', 'true').lower() == 'false':
+            search_params['return_geometry'] = False
 
         try:
-            query_response = self.feature_service_layer.perform_query(**search_params)
-        except ValueError:
+            # Query with limit plus one to determine if the limit excluded any features
+            query_response = self.feature_service_layer.perform_query(int(limit) + 1, offset, **search_params)
+        except (DatabaseError, ValueError):
             return HttpResponseBadRequest(json.dumps({'error': 'Invalid request'}))
 
-        # When requesting selection IDs, ArcGIS sends both returnIdsOnly and returnCountOnly, but expects the response
-        # in the 'returnIdsOnly' format
-        if kwargs.get('returnIdsOnly'):
-            response = {
-                'count': len(query_response),
-                'objectIdFieldName': self.feature_service_layer.object_id_field,
-                'objectIds': [feature[self.feature_service_layer.object_id_field] for feature in query_response]
-            }
-        elif kwargs.get('returnCountOnly', 'false') == 'true':
-            response = {'count': query_response}
+        if return_count_only:
+            response = query_response[0]
         else:
+            record_count = len(query_response)
+            exceeded_limit = record_count > int(limit)
+            query_response = query_response[:-1] if exceeded_limit else query_response
+
             response = {
                 'count': len(query_response),
-                'fields': self.feature_service_layer.fields,
-                'geometryType': 'esriGeometryPoint',
-                'features': covert_wkt_to_esri_feature(query_response)
+                'total': len(query_response),
+                'exceededTransferLimit': exceeded_limit
             }
+            if return_ids_only:
+                object_id_field = self.feature_service_layer.object_id_field
+                response['objectIdFieldName'] = object_id_field
+                response['objectIds'] = [feature[object_id_field] for feature in query_response]
+            else:
+                features = convert_wkt_to_esri_feature(query_response, self.feature_service_layer)
+                response.update({
+                    'count': len(features),        # Root level (source) features
+                    'total': len(query_response),  # Total number of records queried
+                    'fields': self.feature_service_layer.fields,
+                    'geometryType': 'esriGeometryPoint',
+                    'features': features
+                })
+
+                if response['total'] > response['count']:
```

Comment:
You're right, good catch. This was something @brendan-ward asked for: to not include related field data if it's not in the query. I can think of a couple alternatives to making this determination:

1. Search the `outFields` passed in for any prefix that matches a related table (i.e. measurements.*)
2. Loop over all the features returned to see if any have a related items count as determined by `convert_wkt_to_esri_feature`
3. Capture this information when we're iterating over results in `convert_wkt_to_esri_feature`, and somehow send it back
4. A more involved refactor of which function does what, so this one would know what it needs to do a simple check like this one.
5. Just always include the related field definitions

I'm inclined toward option 1 or option 5. I'm becoming concerned about the length and complexity of this method, and the overall complexity of this PR, but I understand how adding field definitions for all related tables could result in a lot of unnecessary baggage.

I almost think, if we want to hone tablo query requests, we need a new issue for it... for instance, maybe we also want to only include source field definitions for queried fields. It seems like we almost need a strategy just for that, and trying to do it here just complicates the number of changes this PR already represents.

Anyway, I'm inclined to just remove this condition and address the concern later. Does anyone object?